api/api.proto

```diff
@@ -2296,3 +2296,32 @@ message ClusterPairsEnumerateResponse {
     // required: false
     map<string, ClusterPairInfo> pairs = 2;
 }
+
+message Catalog {
+	// Name of the Directory/File
+	string name = 1;
+	// Full Path of the Directory/File
+	string path = 2;
+	// Type Directory or File
+	string type = 3;
+	// File or Directory Size
+	int64 size = 4;
+	// Last Modified
+	google.protobuf.Timestamp LastModified = 5;
+	// Children
+	repeated Catalog children = 6;
+}
+
+message Report {
```

Comment:
Report can also be a part of Catalog so that for every sub directory you can report the no. of directories and files in it.

Reply:
This will result in too much data and we won't use it.
build this output after the tree command output